packages/app/src/systems/CRX/background/services/VaultService.ts

```diff
@@ -32,6 +38,18 @@ export class VaultService extends VaultServer {
     this.emitLockEvent();
   }
 
+  async isLocked(): Promise<boolean> {
+    const isWalletLocked = await super.isLocked();
+    if (!isWalletLocked) {
+      const timer = await getTimer();
+      if (timer) {
+        // Saving a new timestamp for wallet auto lock
+        saveTimer(AUTO_LOCK_IN_MINUTES);
```

Comment:
I think we could change the `saveSecret` method logic to save `lockTime` on the side of the `time` on the session. And then use this field value on a method `resetTimer`. This will reduce the need to pass the AUTO_LOCK_IN_MINUTES around and centralize just on `saveSecret`.